aws/rust-runtime/aws-types/src/endpoint.rs

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Newtypes for endpoint-related parameters
+//!
+//! Parameters require newtypes so they have distinct types when stored in layers in config bag.
+
+use aws_smithy_types::config_bag::{Storable, StoreReplace};
+
+/// Newtype for `use_fips`
+#[derive(Clone, Debug)]
+pub struct UseFips(pub bool);
```

Comment:
Would it be nicer to implement `Deref` for these rather than making the inner boolean public so that it could be accessed using `*use_fips`?